File: BackEnd/FoodOrder.Core/ViewModels/OrderDetails/OrderDetailVM.cs
using FoodOrder.Core.Models;
using FoodOrder.Core.ViewModels.Foods;
using FoodOrder.Core.ViewModels.Orders;
using FoodOrder.Core.ViewModels.Ratings;
using FoodOrder.Core.ViewModels.SaleCampaigns;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoodOrder.Core.ViewModels.OrderDetails
{
    public class OrderDetailVM
    {
        public int OrderID { get; set; }
        public int FoodID { get; set; }
        public int Amount { get; set; }
        public double Price { get; set; }
        public int? SaleCampaignID { get; set; }
        public float? SalePercent { get; set; }

        public bool IsDeleted { get; set; }
        public DateTime? TimeDeleted { get; set; }

        public SaleCampaignVM SaleCampaign { get; set; }
        public OrderVM OrderVM { get; set; }
        public FoodVM FoodVM { get; set; }
        public RatingVM RatingVM { get; set; }
    }
}

File: FrontEnd/food_order/lib/view_models/Orders/OrderDetailVM.dart
import 'package:food_delivery/view_models/Foods/FoodVM.dart';
import 'package:food_delivery/view_models/SaleCampaigns/SaleCampaignVM.dart';
import 'package:food_delivery/view_models/ratings/RatingVM.dart';

import 'OrderVM.dart';

import 'package:json_annotation/json_annotation.dart';

part 'OrderDetailVM.g.dart';

@JsonSerializable(explicitToJson: true)
class OrderDetailVM {
  OrderDetailVM();
  late int orderID;
  late int foodID;
  late int amount;
  late double price;
  int? saleCampaignID;
  double? salePercent;

  OrderVM? orderVM;
  FoodVM? foodVM;
  SaleCampaignVM? saleCampaign;
  RatingVM? ratingVM;

  factory OrderDetailVM.fromJson(Map<String, dynamic> json) =>
      _$OrderDetailVMFromJson(json);
  Map<String, dynamic> toJson() => _$OrderDetailVMToJson(this);
}
